src/OrbitVulkanLayer/TimerQueryPool.h

```diff
@@ -104,62 +133,112 @@ class TimerQueryPool {
     return true;
   }
 
-  // Resets an occupied slot to be ready for queries again. It will also call to Vulkan to reset the
-  // content of that slot (in contrast to `RollbackPendingQuerySlots`).
+  // Marks for the given slots that the Vulkan layer will not do any attempts to read the underlying
+  // slots anymore after this call.
+  //
+  // If for a given slot `MarkQuerySlotsForReset` was already called (i.e. `vkResetCommandBuffer`
+  // was called on the command buffers using the slot), it resets an occupied slot to be ready for
+  // queries again. In this case it will also call to Vulkan to reset the content of that slot
+  // (in contrast to `RollbackPendingQuerySlots`).
   //
   // Note that the pool must be initialized using `InitializeTimerQueryPool` before.
   // Further, the given slots must be in the `kReadyForQueryIssue` state, i.e. must be a result
   // of `NextReadyQuerySlot` and must not have been reset yet.
-  void ResetQuerySlots(VkDevice device, const std::vector<uint32_t>& slot_indices) {
-    ResetQuerySlotsInternal(device, slot_indices, false);
+  void MarkQuerySlotsDoneReading(VkDevice device, const std::vector<uint32_t>& slot_indices) {
+    if (slot_indices.empty()) {
+      return;
+    }
+    absl::WriterMutexLock lock(&mutex_);
+    CHECK(device_to_query_slots_.contains(device));
+    std::vector<SlotState>& slot_states = device_to_query_slots_.at(device);
+    CHECK(device_to_free_slots_.contains(device));
+    std::vector<uint32_t>& free_slots = device_to_free_slots_.at(device);
+    for (uint32_t slot_index : slot_indices) {
+      CHECK(slot_index < num_timer_query_slots_);
+      const SlotState& current_state = slot_states.at(slot_index);
+      CHECK(current_state == SlotState::kQueryPendingOnGpu ||
+            current_state == SlotState::kResetRequested);
+      if (current_state == SlotState::kQueryPendingOnGpu) {
+        slot_states.at(slot_index) = SlotState::kDoneReading;
+        continue;
+      }
```

Comment:
Nit: You could just have: 

CHECK(current_state == SlotState::kResetRequested); 

after the "if" clause. I find that a bit easier to read.

Reply:
Agree. Thanks